docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,471 @@
-values, called template parameters.
+You can then define this new instance. The following example declares two
+primitive types named `my_int` and `my_double`, assigns them each a value, and
+includes them in the `my_example` reference type.
 
-You can use stored search templates to run searches without writing out the
-entire query. Just provide the stored template's ID and the template parameters.
-This is useful when you want to run a commonly used query quickly and without
-mistakes.
+[source,painless]
+----
+Example my_example = new Example() <1>
+my_example.my_int = 1; <2>
+my_example.my_double = 2; <3>
+double my_add = my_example.add();
+----
+<1> Create a new instance of `Example`
+<2> Declare `my_int` as part of the `my_example` reference type
+<3> Declare `my_double` as part of the `my_example` reference type
 
-Search templates use the https://mustache.github.io/mustache.5.html[mustache
-templating language]. See <<search-template>> for more information and examples.
+That's a brief overview of how reference types are structured within Painless,
+but what you really need to know is how to use reference types.
 
 [discrete]
-[[modules-scripting-using-caching]]
-=== Script caching
+===== How to use reference types
+What if you want to modify the types and values that are part of a reference
+type? That's where you can leverage the magic of reference types! You can have
+multiple reference type values that point to the _same_ instance, which means
+you don't have to modify the instance itself.
 
-All scripts are cached by default so that they only need to be recompiled
-when updates occur. By default, scripts do not have a time-based expiration, but
-you can change this behavior by using the `script.cache.expire` setting.
-You can configure the size of this cache by using the `script.cache.max_size` setting.
-For most contexts, the default cache size is `100`. For ingest contexts, the
-default cache size is `200`.
+For example, the following statement creates two new reference types:
+`my_example1` and `my_example2`:
 
-NOTE: The size of scripts is limited to 65,535 bytes. This can be
-changed by setting `script.max_size_in_bytes` setting to increase that soft
-limit, but if scripts are really large then a
-<<modules-scripting-engine,native script engine>> should be considered.
+[source,painless]
+----
+Example my_example1 = new Example(); <1>
+Example my_example2 = my_example1; <2>
+my_example2.my_int = 4;
+----
+<1> `my_example1` creates a new instance of `Example`.
+<2> `my_example2` creates a new instance of `my_example1`.
 
-[[scripts-and-search-speed]]
-=== Scripts and search speed
+By creating a new instance of `Example`, `my_example1` references `Example`
+and inherits all of the data and logic that it contains.
 
-Scripts can't make use of {es}'s index structures or related optimizations. This
-can sometimes result in slower search speeds.
+Similarly, `my_example2` references `my_example1` and inherits everything it
```

Comment:
`inherits` is the wrong word here.  They are the same object.  Changing `my_example1.my_int` changes the underlying object.  One would see that in `my_example2.my_int`.